### ISSUE TITLE
Format of binding block changed

### ISSUE THREAD

**Issue by pkqsun** [terraform-providers/terraform-provider-openstack#1008]
Hi there,

### Terraform Version
Terraform v0.12.17
+ provider.openstack v1.24.0

### Affected Resource(s)
Please list the resources as a list, for example:
- openstack_networking_port_v2
- block binding

### Terraform Configuration Files
```hcl
resource "openstack_networking_port_v2" "port_net_15_v4" {
    name = "${var.name_prefix}-${var.hostname}"
    network_id = var.net_15_net_uuid
    admin_state_up = "true"


    binding = {
        vnic_type = "direct"
    }

    fixed_ip {
        subnet_id = var.net_15_v4_sub_uuid
        ip_address = var.net_15_static_ipv4_address
    }

     fixed_ip {
        subnet_id =  var.net_15_v6_sub_uuid
        ip_address = var.net_15_static_ipv6_address
    }

    allowed_address_pairs {
        ip_address = "12.18.22.0/24"
    }

    allowed_address_pairs {
        ip_address = "ipv6:ba:da:22::/64"
    }
}

```

### Debug Output

```
Error: Unsupported argument

  on modules/instance/sbc/group_01/main.tf line 476, in resource "openstack_networking_port_v2" "port_net_11_v4":
 476:     binding = {

An argument named "binding" is not expected here. Did you mean to define a
block of type "binding"?


Error: Unsupported argument

  on modules/instance/sbc/group_01/main.tf line 504, in resource "openstack_networking_port_v2" "port_net_12_v4":
 504:     binding = {

An argument named "binding" is not expected here. Did you mean to define a
block of type "binding"?


Error: Unsupported argument

  on modules/instance/sbc/group_01/main.tf line 538, in resource "openstack_networking_port_v2" "port_net_13_v4":
 538:     binding = {

An argument named "binding" is not expected here. Did you mean to define a
block of type "binding"?
```

### Panic Output
If Terraform produced a panic, please provide a link to a GitHub Gist containing the output of the `crash.log`.

### Expected Behavior
What should have happened?

### Actual Behavior
What actually happened?

### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:
1. `terraform init`
2. `terraform plan/apply`

### Important Factoids
Is there anything atypical about your accounts that we should know? For example: Running in EC2 Classic? A custom version of OpenStack? Tight ACLs?

### References
from https://www.terraform.io/docs/providers/openstack/r/networking_port_v2.html#port-with-physical-binding-information, 

```
resource "openstack_networking_port_v2" "port_1" {
  name           = "port_1"
  network_id     = "${openstack_networking_network_v2.network_1.id}"
  device_id      = "cdf70fcf-c161-4f24-9c70-96b3f5a54b71"
  device_owner   = "baremetal:none"
  admin_state_up = "true"

  binding = {
    host_id   = "b080b9cf-46e0-4ce8-ad47-0fd4accc872b"
    vnic_type = "baremetal"
    profile   = <<EOF
{
  "local_link_information": [
    {
      "switch_info": "info1",
      "port_id": "Ethernet3/4",
      "switch_id": "12:34:56:78:9A:BC"
    },
    {
      "switch_info": "info2",
      "port_id": "Ethernet3/4",
      "switch_id": "12:34:56:78:9A:BD"
    }
  ],
  "vlan_type": "allowed"
}
EOF
  }
}

```

**it is "binding = {}", but indeed, "binding {}" works.**

Please help to clarify the right usage.
BTW, I did not find the change logs for this in https://github.com/terraform-providers/terraform-provider-openstack/blob/master/CHANGELOG.md. So, please point out from which version of provider it has changed.

**Comment by pkqsun** [comment]
kindly ping @kayrus  @jtopjian

**Comment by ozerovandrei** [comment]
Hello @pkqsun.
As far as I understand it was changed while migrating to Terraform v0.12 from v0.11.
We updated tests for port binding in the following commit: https://github.com/terraform-providers/terraform-provider-openstack/commit/edbe4f15d375cc08dd2c21f24603d63a5739cbab
But we haven't updated docs.

I'll fix them.

There is a description of the new syntax: https://www.terraform.io/upgrade-guides/0-12.html#attributes-vs-blocks
